File: e2e_testing_bdd/Pages/SearchPage.cs
using Microsoft.Playwright;

namespace e2e_testing_bdd.Pages;

public class SearchPage
{
    private IPage _page;
    public SearchPage(IPage page) => _page = page;
    
    private ILocator SearchPageTitle => _page.Locator("h1:text(\"Search for a user\")");
    private ILocator SearchInput => _page.Locator("input[name='username']");
    private ILocator SearchButton => _page.Locator("button:text(\"search\")");


    public async Task SearchFor(string searchTerm)
    {
        await SearchPageTitle.WaitForAsync(new LocatorWaitForOptions { State = WaitForSelectorState.Visible });
        await SearchInput.FillAsync(searchTerm);
        await SearchButton.ClickAsync();
    }
}

File: e2e_testing_js/Pages/SearchPage.js
exports.SearchPage = class SearchPage {
    constructor(page) {
        this.page = page;
        this.searchPageTitle = page.locator('h1:text("Search for a user")');
        this.searchInput = page.locator('input[name="username"]');
        this.searchButton = page.locator('button:text("search")');
    }
    
    async searchFor(searchTerm) {
        await this.searchPageTitle.waitFor({ state: 'visible' });
        await this.searchInput.fill(searchTerm);
        await this.searchButton.click();
    }

    async assertSearchButtonBlocked() {
        const isDisabled = await this.searchButton.isDisabled();
        if (!isDisabled) {
            throw new Error('The Search Button should be disable');
        }
    }
}
